lib/worker/src/routes/vaults.rs

```diff
@@ -530,7 +530,16 @@ pub async fn write_event<W: Web3StorageClient>(
         }
     }
 
-    let cid_bytes = match upload_w3s(gcs_client, w3s_client, &filename).await {
+    let mut retries = 0;
+    let cid_bytes = match loop {
```

Comment:
Maybe good to move the retry part into a different function? because this expressions looks quite complicated to read. For example: 

```Rust
pub async fn upload_w3s_with_retry<W: Web3Storage>(
    filename: &String,
    gcs_client: GcsClient,
    w3s_client: W,
) -> basin_common::errors::Result<Vec<u8>> {
    let mut retries = 0;
    loop {
        let result = upload_w3s(gcs_client.clone(), w3s_client.clone(), &filename).await;
        if result.is_ok() || retries > 3 {
            break result;
        } else {
            retries += 1;
            tokio::time::sleep(std::time::Duration::from_millis(100)).await;
        }
    }
}
```

```Rust
 let cid_bytes = match upload_w3s_with_retry(&filename, gcs_client, w3s_client).await {
        .....
    };

```

Reply:
Totally!  This should be in the web3 client actually, transparent to this part of the code, but was having a hard time getting the code to compile haha anyway, this was just a test to see if retrying would fix the problem of uploading big files, but I actually found out what the problem was. I'm going to remove it, actually